src/default_client.rs

```diff
@@ -169,34 +172,70 @@ impl<A: ClientAdapter> Error for DefaultClientError<A> {}
 impl<A: ClientAdapter + Sync + Send> Client for DefaultClient<A> {
     type Error = DefaultClientError<A>;
 
+    /// Add friends by id or username.
+    ///
+    /// Either accept a friend invite or send a friend invite to the specified users
+    /// based on their ids or usernames.
+    ///
+    /// # Example
```

Comment:
One other thing @seanpackham @gpene -- 

Rust documentation has this concept of runnable documentation examples that @marot  is using. They are super cool. Just mentioning since I know it's on your guys radar to think about how to consolidate/systematize docs.

Reply:
@lugehorsam in that case we can duplicate some documentation. If there's a accepted practice amongst Rust developers lets follow that and also cover the same concepts as our other client libraries in our documentation.